tests/models/test_aggregate.py

```diff
@@ -512,11 +503,21 @@ def aggregate_nd(nd):
     d_in = dt.Frame(matrix)
     d_in_copy = dt.Frame(d_in)
 
+    messages = []
+    def progress_fn(p):
+        assert 0 <= p.progress <= 1
+        assert p.status in ("running", "finished", "cancelled", "error")
```

Comment:
Guess we also should add this check here
```
assert p.message in ("Preparing", "Aggregating", "Sampling", "Finalizing")
```

and this one in the bottom
```
assert messages[0].message == "Preparing"
assert messages[-1].message == "Finalizing"
```
to make sure the messages are passed correctly to the `progress_fn`.

Reply:
Btw, is `.message` is the same thing that is showing up right next to the progress bar?

Reply:
Yes, it's the same thing

Reply:
Ok, I will update this test shortly then.

Reply:
When I print it, It looks like quite a lot of initial messages and the final one are empty strings:

```
[]
...
[]
[Aggregating]
...
[Finalizing]
[]
```

Reply:
Yes, there is no message at "finished" state, and also we did not setup any message at the initial stages of the job.